File: src/main/java/com/copareplay/api/entity/WorldCup2026Result.java
package com.copareplay.api.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "world_cup_2026_results")
public class WorldCup2026Result {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String homeTeam;

    @Column(nullable = false)
    private String awayTeam;

    private LocalDateTime matchDate;

    @Column(nullable = false)
    private String stage;

    private String groupName;
    private String stadium;
    private String city;

    @Column(unique = true)
    private String sourceMatchId;

    @Column(nullable = false)
    private Boolean finished;

    @Column(nullable = false)
    private Integer homeScore;

    @Column(nullable = false)
    private Integer awayScore;

    @Column(length = 500)
    private String note;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = createdAt;
        if (finished == null) {
            finished = false;
        }
    }

    @PreUpdate
    void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: copa-replay-mobile/src/types/worldCup2026Result.ts
export type WorldCup2026Result = {
  id: number;
  homeTeam: string;
  awayTeam: string;
  matchDate?: string | null;
  stage: string;
  groupName?: string | null;
  stadium?: string | null;
  city?: string | null;
  sourceMatchId?: string | null;
  finished: boolean;
  homeScore: number;
  awayScore: number;
  note?: string | null;
  createdAt: string;
  updatedAt: string;
};

export type WorldCup2026ResultPayload = {
  homeTeam: string;
  awayTeam: string;
  matchDate?: string | null;
  stage: string;
  groupName?: string;
  stadium?: string;
  city?: string;
  finished?: boolean;
  homeScore: number;
  awayScore: number;
  note?: string;
};
